nemo/collections/asr/models/ssl_models.py

```diff
@@ -554,9 +554,11 @@ def validation_step(self, batch, batch_idx, dataloader_idx=0):
         self.reset_registry()
         del self._in_validation_step
 
-        return {
-            'val_loss': loss_value,
-        }
+        val_log_dict = {'val_loss': loss_value}
+
+        self.log_dict(val_log_dict)
+
+        return val_log_dict
 
     def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
         val_loss_mean = torch.stack([x['val_loss'] for x in outputs]).mean()
```

Comment:
Is this one the loss value that should be logged? the one in validation_step is per-batch, but we need to monitor the mean loss over all validation data

Reply:
The logging has to be done at validation step itself akin to the change introduced in this PR for PTL upgrade - https://github.com/NVIDIA/NeMo/pull/6433/files#diff-b2780d88910b132d177fb0081453ad276c5e4aefe47a87f219e96f38af0625be

Reply:
If we do the logging at multi_validation_epoch_end and multi_test_epoch_end, we still get the current error - `ModelCheckpoint(monitor='val_wer')` could not find the monitored key in the returned metrics: ['train_loss', 'learning_rate', 'global_step', 'train_backward_timing in s', 'train_step_timing in s', 'training_batch_wer', 'epoch', 'step']. HINT: Did you call `log('val_wer', value)` in the `LightningModule`?

Reply:
I'm refactoring the PR currently to make the logging similar to how we do it for ctc_models. I'll make the change for RNNT and Hybrid models for now, maybe we can open another PR next to address these issues for the SLU, SSL and label models

Reply:
New PR - https://github.com/NVIDIA/NeMo/pull/7531